tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+        p.spawn(run_fio, **fio_ready(config, cache_client))
+        p.spawn(
+            pwl.check_cache_file_exists,
+            image_spec,
+            config["fio"].get("runtime", 120),
+            **config,
+        )
+
+
+def restart_service_with_cache_test(
+    service_name, rados_obj, pwl, config, cache_client, image_spec
+):
+    """
+    Restart a Ceph service and verify the persistent write cache.
+
+    Args:
+        service_name (str): The name of the Ceph service to restart.
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(f"Test to restarting {service_name} service along with pwl cache test")
+    with parallel() as p:
+        p.spawn(rados_obj.restart_daemon_services, service_name)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def restart_osd_service_with_cache_test(
+    cache_client, rados_obj, pwl, config, image_spec
+):
+    """
+    Restart OSD service and verify the persistent write cache.
+
+    Args:
+        cache_client (Node): The client node for cache testing.
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart OSD service along with pwl cache test")
+
+    def restart_osd():
+        cmd = "ceph orch ls osd"
+        out = cache_client.exec_command(cmd=cmd, output=True)
+
+        # Regular expression to extract the first OSD name
+        osd_daemon = re.search(r"osd\.(\S+)", str(out))
+        cmd = f"ceph orch restart {osd_daemon[0]}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+    with parallel() as p:
+        p.spawn(restart_osd)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def restart_ceph_target_with_cache_test(
+    admin_node, pwl, config, cache_client, image_spec
+):
+    """
+    Restart Ceph target service and verify the persistent write cache.
+
+    Args:
+        admin_node (Node): The admin node for restarting the ceph.target service.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart ceph target service along with pwl cache test")
+
+    def restart_ceph_target():
+        cmd = "systemctl restart ceph.target"
+        admin_node.exec_command(cmd=cmd, sudo=True)
+
+    with parallel() as p:
+        p.spawn(restart_ceph_target)
```

Comment:
This implementation still doesn't catch any return value and throw exception right? As we discussed.